sklego/preprocessing.py

```diff
@@ -609,3 +610,133 @@ def _check_X_for_type(X):
         """Checks if input of the Selector is of the required dtype"""
         if not isinstance(X, pd.DataFrame):
             raise TypeError("Provided variable X is not of type pandas.DataFrame")
+
+class RepeatingBasisFunction(TransformerMixin, BaseEstimator):
+    """
+    This is a transformer for features with some form of circularity. 
+    E.g. for days of the week you might face the problem that, conceptually, day 7 is as 
+    close to day 6 as it is to day 1. While numerically their distance is different.
+    This transformer solves that problem.
+    The transformer selects a column and transforms it with a give number of repeating (radial) basis functions.
+    Radial basis functions are bell-curve shaped functions which take the original data as input.
+    The basis functions are equally spaced over the input data range (specified with floor and ceil).
+    The key feature of repeating basis funtions is that they are continuous when moving
+    from the min to the max. As a result these repeating basis functions can capture the 
+    the "close" to the center of each respective basis function
+
+    :type column: int or list, default=0
+    :param column: Indexes the data on its second axis. Integers are interpreted as
+        positional columns, while strings can reference DataFrame columns by name.
+    
+    :type remainder: {'drop', 'passthrough'}, default="drop"
+    :param remainder: By default, only the specified column is transformed, and the 
+        non-specified columns are dropped. (default of ``'drop'``). By specifying 
+        ``remainder='passthrough'``, all remaining columns will be automatically passed 
+        through. This subset of columns is concatenated with the output of the transformer.
+
+    :type n_periods: int, default=12
+    :param n_periods: number of basis functions to create, i.e., the number of columns that
+        will exit the transformer.
+
+    :type floor: float or 'min', default='min'
```

Comment:
I am a bit unsure what you mean with the floor/ceil parameters here. The RBF spans between 0-1 no? 

![image](https://user-images.githubusercontent.com/1019791/62199297-841b9a80-b383-11e9-9a89-256d84958763.png)

Reply:
Thanks. Two actionable points of feedback come out of this:
1. Ensure the curves always range between 0 and 1. Now this is note the case, but that would be neater.
2. Clearly label what ceil/floor do. What they do is scale the input data. Imagine you want to make repeating basis functions on a column with hour of the day. But your training data does not include data points between midnight and 3am. You might still want to have your repeating basis functions be split on the 0-24 range rather than the 3-24 range. Explicitly saying what the minimum and maximum of your input range fixes this.

Will take them on board next week

Reply:
I've renamed floor and ceil to one parameter `input_range` which accepts a tuple. I hope it now better reflects what it does: it captures at what values the input data starts over again. E.g. for days of the week you'd provide (1,7)